test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -70,26 +74,43 @@
 
 using testing::_;
 using testing::HasSubstr;
+using testing::NiceMock;
+using testing::Ref;
 using testing::Return;
 
 namespace quic {
 namespace {
-
 class QuicPlatformTest : public testing::Test {
 protected:
   QuicPlatformTest()
-      : log_level_(GetLogger().level()), verbosity_log_threshold_(GetVerbosityLogThreshold()) {
+      : log_level_(GetLogger().level()), verbosity_log_threshold_(GetVerbosityLogThreshold()),
+        symbol_table_(Envoy::Stats::SymbolTableCreator::makeSymbolTable()), alloc_(*symbol_table_),
+        store_(std::make_unique<Envoy::Stats::ThreadLocalStoreImpl>(alloc_)) {
     SetVerbosityLogThreshold(0);
     GetLogger().set_level(ERROR);
+    store_->addSink(sink_);
+  }
+
+  void SetUp() override {
+    QuicStatsImpl& stats = getQuicheStats();
+    stats.initializeStats(*store_);
   }
 
+  void TearDown() override { getQuicheStats().resetForTest(); }
+
   ~QuicPlatformTest() override {
     SetVerbosityLogThreshold(verbosity_log_threshold_);
     GetLogger().set_level(log_level_);
   }
 
   const QuicLogLevel log_level_;
   const int verbosity_log_threshold_;
+  Envoy::Stats::SymbolTablePtr symbol_table_;
+  NiceMock<Envoy::Event::MockDispatcher> main_thread_dispatcher_;
+  NiceMock<Envoy::ThreadLocal::MockInstance> tls_;
+  Envoy::Stats::AllocatorImpl alloc_;
+  Envoy::Stats::MockSink sink_;
+  std::unique_ptr<Envoy::Stats::ThreadLocalStoreImpl> store_;
```

Comment:
for the test, see if you can use Envoy::Stats::TestUtil::TestStore. That will allow you in the test to do string based lookups of stats that have mixed dynamic and symbolic name tokens.

Reply:
Good point: it would be better if we can actually test real histogram functionality with TestStore. Could you instead just simply add deliverHistogramToSinks into TestStore itself rather than overriding?